engine/collection/proposal/engine.go

```diff
@@ -180,25 +180,31 @@ func (e *Engine) Process(originID flow.Identifier, event interface{}) error {
 // SendVote will send a vote to the desired node.
 func (e *Engine) SendVote(blockID flow.Identifier, view uint64, sigData []byte, recipientID flow.Identifier) error {
 
+	log := e.log.With().
+		Hex("collection_id", blockID[:]).
+		Uint64("collection_view", view).
+		Hex("recipient_id", recipientID[:]).
+		Logger()
+	log.Info().Msg("processing vote transmission request from hotstuff")
+
 	// build the vote message
 	vote := &messages.ClusterBlockVote{
 		BlockID: blockID,
 		View:    view,
 		SigData: sigData,
 	}
 
-	err := e.conduit.Unicast(vote, recipientID)
-	if err != nil {
-		return fmt.Errorf("could not send vote: %w", err)
-	}
-
-	e.log.Debug().
-		Hex("block_id", blockID[:]).
-		Uint64("view", view).
-		Hex("recipient_id", recipientID[:]).
-		Msg("sending vote")
-
-	e.engMetrics.MessageSent(metrics.EngineProposal, metrics.MessageClusterBlockVote)
+	// TODO: this is a hot-fix to mitigate the effects of the following Unicast call blocking occasionally
+	go func() {
```

Comment:
Better to use `e.unit.Launch` so that the lifecycle of goroutine is the same as the engine itself.